File: CPP/No492.cpp
/**
 * Created by Xiaozhong on 2020/10/2.
 * Copyright (c) 2020/10/2 Xiaozhong. All rights reserved.
 */

#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

class Solution {
public:
    vector<int> constructRectangle(int area) {
        int sideLength = sqrt(area);
        if (sideLength * sideLength == area) return {sideLength, sideLength};
        for (int i = sideLength; i >= 1; --i) {
            if (area % i == 0) {
                return {area / i, i};
            }
        }
        return {};
    }
};

int main() {
    Solution s;
    cout << s.constructRectangle(10)[0] << " - " << s.constructRectangle(10)[1] << endl;
}

File: Java/src/No492.java
/*
 * Copyright (c) 10/23/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

import java.util.Arrays;

public class No492 {
    public int[] constructRectangle(int area) {
        int[] ans = {};
        int mid = (int) Math.sqrt(area);
        for (int i = 1; i <= mid; ++i) {
            if (area % i == 0) ans = new int[]{area / i, i};
        }
        return ans;
    }

    public static void main(String[] args) {
        No492 no492 = new No492();
        System.out.println(Arrays.toString(no492.constructRectangle(4)));
    }
}
